staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -963,3 +965,79 @@ func SelectorFromValidatedSet(ls Set) Selector {
 func ParseToRequirements(selector string, opts ...field.PathOption) ([]Requirement, error) {
 	return parse(selector, field.ToPath(opts...))
 }
+
+// SetSelector wraps a Set, allowing it to implement the Selector interface. Unlike
+// Set.AsSelectorPreValidated (which copies the input Set), this type simply wraps the underlying
+// Set. As a result, it is substantially more efficient, but requires the caller to not mutate the
+// Set.
```

Comment:
might be worth noting that none of the SetSelector method implementations mutate the set, and that the Add() and Requirements() methods convert to a less efficient selector implementation